### PR TITLE
Automatic version updates for v1.36.1 release

### PR DESCRIPTION
**Pull request by danudey**
Updates for Calico Enterprise v3.20.0-2.0.

### CODE REVIEW

File: config/enterprise_versions.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Components defined here are required to be kept in sync with hack/gen-versions/enterprise.go.tpl
-title: release-calient-v3.20
+title: v3.20.0-2.0
```
Comment (rene-dekker) [inline config/enterprise_versions.yml:2]: Shouldn't all versions in this file be changed to v3.20.0-2.0?

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Pull request merged**